finality-aleph/src/party/mod.rs

```diff
@@ -372,17 +402,19 @@ where
         node_id: NodeIndex,
         authorities: Vec<AuthorityId>,
     ) -> AuthorityTask {
-        let keybox = KeyBox::new(
-            node_id,
-            AuthorityVerifier::new(authorities.clone()),
+        let authority_verifier = AuthorityVerifier::new(authorities.clone());
+        let authority_pen =
             AuthorityPen::new(authorities[node_id.0].clone(), self.keystore.clone())
                 .await
-                .expect("The keys should sign successfully"),
-        );
+                .expect("The keys should sign successfully");
+
+        let keybox = KeyBox::new(node_id, authority_verifier.clone(), authority_pen.clone());
+
+        //TODO: we should retry
```

Comment:
Same comment as above about perhaps returning the error, but at least remove the todo and create a task.

Reply:
Done